profiling/stack-collector/stack-collector.c

```diff
@@ -0,0 +1,101 @@
+#include "stack-collector.h"
+
+#include <Zend/zend_compile.h>
+#include <Zend/zend_portability.h>
+
+typedef datadog_php_string_view string_view_t;
+
+void datadog_php_stack_collect(zend_execute_data *execute_data,
+                               datadog_php_stack_sample *sample) {
```

Comment:
This is a perfect candidate to mark as `DATADOG_HOT`.

```c
#define DATADOG_HOT __attribute__((hot))
```

Not a blocker for this PR, but as we discussed, once we get some direct tests set up in ZAI SAPI we can comb through this more throughly for micro optimizations.

Mmmmmm, `DATADOG_HOT` makes me think of 🌭. 😋

Reply:
I added `DDTRACE_HOT` and `DDTRACE_COLD`, and migrated existing `ZEND_COLD` usage to `DDTRACE_COLD`, and added it to a few more places while I was at it.

Reply:
Awesome! It looks like `datadog_php_stack_collect` wasn't marked with `DDTRACE_HOT`.